shared/bitutil/bit.go

```diff
@@ -83,3 +78,20 @@ func fillBit(target byte, index uint64) byte {
 	bitShift := 7 - index
 	return target ^ (1 << bitShift)
 }
+
+// BitfieldBit extracts the bit in bitfield at position i.
+// Spec pseudocode definition:
+//   def get_bitfield_bit(bitfield: bytes, i: int) -> int:
+//     """
+//     Extract the bit in ``bitfield`` at position ``i``.
+//     """
+//     return (bitfield[i // 8] >> (i % 8)) % 2
+func BitfieldBit(bitfield []byte, i int) byte {
+	//take the relevant byte in bitfield in order to get i value
```

Comment:
let's add a space here for consistency
```suggestion
	// take the relevant byte in bitfield in order to get i value
```